app/actions/remote/channel.ts

```diff
@@ -459,3 +459,44 @@ export const switchToChannelByName = async (serverUrl: string, channelName: stri
         return {error};
     }
 };
+
+export const fetchChannelInfo = async (serverUrl: string, channelId: string, fetchOnly = false) => {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+        const channels = await queryChannelsById(database.database, [channelId]);
+        let purpose = '';
+        let header = '';
+        const channelModel = channels?.[0];
+        if (channelModel) {
+            const oldInfo = await channelModel.info.fetch();
+            purpose = oldInfo.purpose;
+            header = oldInfo.header;
+        } else {
+            const channel = await client.getChannel(channelId);
```

Comment:
I wonder how often we will find this situation, and whether we should store the channel as well.

Reply:
I would say we don't need to store the channel as we already have it, actually I don't quite understand why we are doing all this, perhaps you can expand.. I know this is a thing on v1, but do we need it?

Reply:
There is a mismatch between where the information is in the server and where it is in the database, and it is trying to deal with that mismatch.

If we ask the server, the purpose and the header of the channel are in the channel object.
If we ask the database, the purpose and the header of the channel are in the channel info.

We could simplify this very much by changing the database and make it work the same in server and client, but not sure why this was done this way in the first place.

Reply:
The answer is super simple to avoid unnecesary re-renders as the header and purpose are only visible in the channel info screen.

Reply:
In that case, the "problem" here is that we are fetching the channel info, and we want to make sure we have the correct purpose and channel header. If we have the channel in the database, we take the info, and set the header and purpose based on that. (Probably here we should query directly the info, instead of querying first the channel. I will solve that.)

If we don't have the channel in the database (which is the situation I don't know how often will happen), I guess we shouldn't leave the purpose and header blank, so we get the channel form the server, and fill that information.

Finally, we do what we came here for, fetch the stats, and create with the stats, the purpose and the header the channel info object to add to the database.

Reply:
Ok that makes sense, but the question is why would we care if we don't have the channel already?

Reply:
The only reason we would care is because we are directly asking for that information (we are calling `fetchChannelInfo`). Here comes again my thought of how often we will call `fetchChannelInfo` about a channel we don't have in the database.

One thing that comes to mind is when we are batching actions. We fetch channel A with fetchOnly, and then fetch its info (also with fetchOnly), and then batch all the database updates together. In that case, when we execute fetchChannelInfo, the channel is still not in the database. But maybe for such case we may want a function like "fetchChannelAndInfo".

Reply:
alright alright, I think I know what is going on and what the confusion is.

When we fetch a channel we split that and store the channel and the channelInfo, then if we fetch stats we would update the channelInfo record in the database.

When you fetch myChannelMember, we use that to store the MyChannel, ChannelMembership and MyChannelSettings.

in fact, if you look at `fetchMyChannel` you'll see how al that is handled.